### PR TITLE
pymatgen 

### PR DESCRIPTION
**Pull request by jan-janssen**
@conda-forge/help-python-c

I am aware there are other pull requests for pymatgen but all of them seem to be inactive: 

- https://github.com/conda-forge/staged-recipes/pull/7413
- https://github.com/conda-forge/staged-recipes/pull/2158
- https://github.com/conda-forge/staged-recipes/pull/2157
- https://github.com/conda-forge/staged-recipes/pull/2013
- https://github.com/conda-forge/staged-recipes/pull/2010

This can only work, once https://github.com/conda-forge/monty-feedstock/pull/12 is merged. 

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Ps: I just noticed this is PR https://github.com/conda-forge/staged-recipes/pull/8888 - I guess that means it is time to celebrate.

### CODE REVIEW

File: recipes/pymatgen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    - feff_plot_cross_section = pymatgen.cli.feff_plot_cross_section:main
+    - feff_plot_dos = pymatgen.cli.feff_plot_dos:main
+    - gaussian_analyzer = pymatgen.cli.gaussian_analyzer:main
+    - get_environment = pymatgen.cli.get_environment:main
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - setuptools
+    - numpy
```
Comment (scopatz) [inline recipes/pymatgen/meta.yaml]: is numpy really a build dep?

Comment (jan-janssen) [reply recipes/pymatgen/meta.yaml]: No, I removed it - thanks for the feedback.

**Comment by jan-janssen** [comment]
@scopatz I fixed the build script. From my perspective it is now ready to be merged.

File: recipes/pymatgen/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    - matplotlib >=1.5
+    - palettable >=3.1.1
+    - sympy
+    - pandas
+    - pybtex
+    - apscheduler >=2.1.0
+    - netcdf4
+    - ase >=3.3
+    - vtk >=6.0.0
+    - dataclasses >=0.6  # [py<37]
```
Comment (scopatz) [inline recipes/pymatgen/meta.yaml]: are all of the above really host requirements? it seems like most of them are only needed at run time.

File: recipes/pymatgen/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  run:
+    - python
+    - cython
```
Comment (scopatz) [inline recipes/pymatgen/meta.yaml]: Is cython a run time requirement?

File: recipes/pymatgen/meta.yaml (hunk 2 above)
Comment (jan-janssen) [reply recipes/pymatgen/meta.yaml]: I removed the ones listed in `extras_require`, the rest is listed under `install_require` 
https://github.com/materialsproject/pymatgen/blob/master/setup.py

File: recipes/pymatgen/meta.yaml (hunk 3 above)
Comment (jan-janssen) [reply recipes/pymatgen/meta.yaml]: You're right, I guess I messed this up during debugging.

File: recipes/custodian/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - cstdn = custodian.cli.cstdn:main
+    - run_vasp = custodian.cli.run_vasp:main
+    - run_nwchem = custodian.cli.run_nwchem:main
+    - converge_kpoints = custodian.cli.converge_kpoints:main
+    - converge_geometry = custodian.cli.converge_geometry:main
+
+requirements:
+  build:
```
Comment (isuruf) [inline recipes/custodian/meta.yaml]: `build` -> `host`

File: recipes/pymatgen/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 2872f431cb1e9dd64e52f89ec669050ea7e90d2acdfcad4d1fc25b1e1d6e8b19
+
+build:
+  skip: true  # [win or py<35]
+  number: 0
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record record.txt"
```
Comment (isuruf) [inline recipes/pymatgen/meta.yaml]: Why not use the `pip` line as in the example?

File: recipes/pymatgen/LICENSE.rst [hunk 6]
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
```
Comment (isuruf) [inline recipes/pymatgen/LICENSE.rst]: This file is unnecessary. conda-build will find it from the source tarball

File: recipes/custodian/LICENSE [hunk 7]
```diff
@@ -0,0 +1,19 @@
+The MIT License (MIT)
```
Comment (isuruf) [inline recipes/custodian/LICENSE]: Same here

File: recipes/custodian/meta.yaml (hunk 4 above)
Comment (jan-janssen) [reply recipes/custodian/meta.yaml]: @isuruf But this breaks the build process - so I would revert your commit: 
`RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.org/simple/numpy/`

File: recipes/pymatgen/meta.yaml (hunk 5 above)
Comment (jan-janssen) [reply recipes/pymatgen/meta.yaml]: I was following the recommendation in https://github.com/conda-forge/staged-recipes/wiki/Frequently-asked-questions (8)

**Comment by isuruf** [comment]
@jan-janssen, that issue with setuptools building was because of the two recipes here. Can you please send a new PR for custodian?

**Comment by isuruf** [comment]
Issue is https://github.com/conda-forge/staged-recipes/issues/6167

File: recipes/pymatgen/meta.yaml (hunk 5 above)
Comment (isuruf) [reply recipes/pymatgen/meta.yaml]: I removed that recommendation. See https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

**Comment by jan-janssen** [comment]
> @jan-janssen, that issue with setuptools building was because of the two recipes here. Can you please send a new PR for custodian?

Done in https://github.com/conda-forge/staged-recipes/pull/8893

**Comment by jan-janssen** [comment]
@isuruf and @scopatz I tried to include your feedback. Is it possible to merge the pull request now?

**Comment by scopatz** [comment]
Thanks!

File: recipes/pymatgen/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - setuptools >=18.0
+    - cython
+    - numpy >=1.14.3
```
Comment (isuruf) [inline recipes/pymatgen/meta.yaml]: This is wrong. This should be `numpy 1.14.3`

**Pull request merged**